rskj-core/src/main/java/co/rsk/net/SnapshotProcessor.java

```diff
@@ -73,8 +83,11 @@ public class SnapshotProcessor implements InternalService, SnapProcessor {
     public static final int BLOCK_CHUNK_SIZE = 400;
     public static final int BLOCKS_REQUIRED = 6000;
     public static final long CHUNK_ITEM_SIZE = 1024L;
+    public static final String TMP_NODES_DIR_NAME = "snapSyncTmpNodes";
     private final Blockchain blockchain;
     private final TrieStore trieStore;
+    private KeyValueDataSource tmpSnapSyncKeyValueDataSource;
```

Comment:
can it be SnapSyncState where we store this db instance, as the lifecycle of it is limited by the time of snap syncing? On the other hand, SnapshotProcessor stays alive until node shutdown. Alternatively, you could manage lifetime of tmpSnapSyncKeyValueDataSource in this class and close it right after snap finishes..

Reply:
I think snapshotprocessor is fine because we should be able to keep the connection alive until snapshot processor stops working. We need to consider that when a node is fully sync, if it stays behind, snap sync will happen again, so we don't need to recreate the folder structure again and start the connection again

Reply:
Addressed